core-kotlin-modules/core-kotlin-date-time/src/test/kotlin/com/baeldung/timedifference/TimePeriodDifferenceUnitTest.kt

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.timedifference
+
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.TimeSource
+import kotlinx.datetime.Instant
+
+class TimePeriodDifferenceUnitTest {
+
+    @Test
+    fun `given two duration periods, when find difference using minus method, then give correct duration value`() {
+        val startDuration = 1.days + 12.hours + 15.minutes
+        val endDuration = 8.hours + 45.minutes
+        val difference: Duration = startDuration.minus(endDuration)
+        val days: Long = difference.inWholeDays
+        val hours: Long = difference.inWholeHours - days.days.inWholeHours
+        val mins: Long = difference.inWholeMinutes - days.days.inWholeMinutes - hours.hours.inWholeMinutes
+        assertEquals(1L, days)
+        assertEquals(3L, hours)
+        assertEquals(30L, mins)
+    }
+
+    @Test
+    fun `given two time marks, when find difference, then give correct time difference`() {
+        val timeSource = TimeSource.Monotonic
+        val mark1 = timeSource.markNow()
+        Thread.sleep(1000) // Sleep 0.5 seconds.
+        val mark2 = timeSource.markNow()
+        val diff = mark2 - mark1
+        val seconds: Long = diff.inWholeSeconds
+        assertEquals(1L, seconds)
+    }
+
+    @Test
+    fun `given two epoch values, when find difference using Instant creation, then give correct duration value`() {
+        val startEpochMillis: Long = 1700617592000L
+        val endEpochMillis: Long = 1700617692000L
```

Comment:
It's already clear the values are Long from the L suffix
```suggestion
        val startEpochMillis = 1700617592000L
        val endEpochMillis = 1700617692000L
```